src/core/config.go

```diff
@@ -435,6 +437,19 @@ func (config *Configuration) GetBuildEnv() []string {
 	return config.buildEnvStored
 }
 
+// GetPassEnv returns environment variables from the user's shell.
+func (config *Configuration) GetPassEnv() []string {
```

Comment:
Would be nice to add this to GetBuildEnv as well so it only has to build the slice once.

They should also form part of the config hash which that will take care of.

Reply:
Just combined into the same function, something like this?
```go
// GetBuildEnv returns the build environment configured for this config object.
func (config *Configuration) GetBuildEnv() []string {
	config.buildEnvOnce.Do(func() {
		config.buildEnvStored = []string{}

		// from the BuildEnv config keyword
		for k, v := range config.BuildEnv {
			pair := strings.Replace(strings.ToUpper(k), "-", "_", -1) + "=" + v
			config.buildEnvStored = append(config.buildEnvStored, pair)
		}

		// from the user's environment based on the PassEnv config keyword
		for _, k := range config.Build.PassEnv {
			v, isSet := os.LookupEnv(k)
			if isSet {
				config.buildEnvStored = append(config.buildEnvStored, k+"="+v)
			}
		}

		sort.Strings(config.buildEnvStored)
	})
	return config.buildEnvStored
}
```

Reply:
Yup, exactly!